go/coinstacks/cosmos/api/handlers.go

```diff
@@ -90,23 +91,57 @@ func (h *Handler) StartWebsocket() error {
 	return nil
 }
 
-func (h *Handler) GetInfo() (api.Info, error) {
-	totalSupply, err := h.httpClient.GetTotalSupply("uatom")
-	if err != nil {
-		return nil, err
-	}
+func (h *Handler) GetRawAprData() (string, string, string, string, error) {
+	var anyError error
 
-	annualProvisions, err := h.httpClient.GetAnnualProvisions()
-	if err != nil {
-		return nil, err
-	}
+	var totalSupply string
+	var annualProvisions string
+	var communityTax string
+	var bondedTokens string
+	var wg sync.WaitGroup
 
-	communityTax, err := h.httpClient.GetCommunityTax()
-	if err != nil {
-		return nil, err
-	}
+	wg.Add(4)
+	go func() {
+		var err error
+		totalSupply, err = h.httpClient.GetTotalSupply("uatom")
+		if err != nil {
+			anyError = err
+		}
+		wg.Done()
+	}()
+	go func() {
+		var err error
+		annualProvisions, err = h.httpClient.GetAnnualProvisions()
+		if err != nil {
+			anyError = err
+		}
+		wg.Done()
+
+	}()
+	go func() {
+		var err error
+		communityTax, err = h.httpClient.GetCommunityTax()
+		if err != nil {
+			anyError = err
+		}
+		wg.Done()
+
+	}()
+	go func() {
+		var err error
+		bondedTokens, err = h.httpClient.GetBondedTokens()
+		if err != nil {
+			anyError = err
+		}
+		wg.Done()
 
-	bondedTokens, err := h.httpClient.GetBondedTokens()
+	}()
+	wg.Wait()
+	return totalSupply, annualProvisions, communityTax, bondedTokens, anyError
+}
+
+func (h *Handler) GetInfo() (api.Info, error) {
+	totalSupply, annualProvisions, communityTax, bondedTokens, err := h.GetRawAprData()
 	if err != nil {
 		return nil, err
```

Comment:
are we intentionally doing all or nothing? if `GetUnbondings` fails but everything else succeeds should we hold out on the rest of the data?

Reply:
Keeps it simple in my mind. Its gonna be so rare that any of them fail and the rest succeed